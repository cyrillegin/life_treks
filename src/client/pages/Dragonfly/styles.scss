$light-green: #dcffdc;

.dragonfly-root {
  margin: 5rem auto;
  width: 70%;

  .description-container {
    line-height: 1.6em;
    text-align: justified;
  }

  img {
    float: right;
    margin: 24px 0 0 14px;
    width: 53%;
  }

  .demo-container {
    display: flex;
    flex-grow: 1;
    height: 750px;
    margin-bottom: 4rem;
  }

  .full-paper {
    padding: 16px;
    transition: 500ms;
    width: 100%;
  }

  .left-paper {
    padding: 16px;
    transition: 500ms;
    width: 33%;
  }

  .right-paper {
    margin-left: 16px;
    padding: 16px;
    transition: 500ms;
    width: 100%;
  }

  .text-field {
    width: 100%;
  }

  .selected-plugin {
    background: $light-green;
  }
}
