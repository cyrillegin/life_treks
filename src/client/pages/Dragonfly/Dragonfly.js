import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MonetizationOnIcon from '@material-ui/icons/MonetizationOn';
import ACUnitIcon from '@material-ui/icons/AcUnit';
import ShowChartIcon from '@material-ui/icons/ShowChart';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import DialogTitle from '@material-ui/core/DialogTitle';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContentText from '@material-ui/core/DialogContentText';

const styles = theme => ({
  root: {
    width: screen.width > 850 ? '80%' : '90%',
    margin: 'auto',
  },
  descriptionContainer: {
    textAlign: 'justified',
    lineHeight: '1.6em',
  },

  image: {
    width: '53%',
    float: 'right',
    margin: '24px 0 0 14px',
  },
  demoContainer: {
    flexGrow: 1,
    display: 'flex',
  },
  fullPaper: {
    width: '100%',
    padding: '16px',
    transition: '500ms',
  },
  leftPaper: {
    width: '33%',
    padding: '16px',
    transition: '500ms',
  },
  rightPaper: {
    width: '100%',
    marginLeft: '16px',
    padding: '16px',
    transition: '500ms',
  },
  textField: {
    width: '100%',
  },
  selectedPlugin: {
    background: '#dcffdc',
  },
});

export class DragonflyPage extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    testPlugin: PropTypes.func.isRequired,
    plugins: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  };

  constructor() {
    super();
    this.state = {
      dialogIsOpen: false,
      dialogMessage: '',
      dialogTitle: '',
      selectedPlugin: '',
      pollRate: '',
      name: '',
      description: '',
      coefficients: '',
      station: '',
      pin: '',
      units: '',
      endpoint: '',
      meta: '',
    };
  }

  handlePluginSelect(name) {
    this.setState({
      selectedPlugin: name,
    });
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  };

  render() {
    const testSensor = () => {
      this.props
        .testPlugin(this.state.selectedPlugin, {
          name: this.state.name,
          description: this.state.description,
          coefficients: this.state.coefficients,
          station: this.state.station,
          pollRate: this.state.pollRate,
          pin: this.state.pin,
          units: this.state.units,
          endpoint: this.state.endpoint,
          meta: this.state.meta,
        })
        .then(data => {
          this.setState({
            dialogIsOpen: true,
            dialogMessage: data,
            dialogTitle: 'Test Result',
          });
        });
    };

    const closeDialog = () => {
      this.setState({
        dialogIsOpen: false,
      });
    };

    const pollerIcon = icon => {
      switch (icon) {
        case 'cryptoPoller':
          return <MonetizationOnIcon />;
        case 'gpioPoller':
          return <ACUnitIcon />;
        default:
          return <ShowChartIcon />;
      }
    };

    const fullscreen = e => {
      const image = document.getElementById('dragonfly-image');
      if (image.requestFullscreen) {
        image.requestFullscreen();
      } else if (image.webkitRequestFullscreen) {
        image.webkitRequestFullscreen();
      } else if (image.mozRequestFullScreen) {
        image.mozRequestFullScreen();
      } else if (image.msRequestFullscreen) {
        image.msRequestFullscreen();
      } else {
        console.error('Fullscreen API is not supported.');
      }
    };

    return (
      <div className={this.props.classes.root}>
        <div className={this.props.classes.demoContainer}>
          <Dialog
            open={this.state.dialogIsOpen}
            onClose={closeDialog}
            onClick={event => {
              event.nativeEvent.stopImmediatePropagation();
            }}
          >
            <DialogTitle>{this.state.dialogTitle}</DialogTitle>
            <DialogContent>
              <DialogContentText>{this.state.dialogMessage}</DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button onClick={closeDialog} color="primary">
                Okay
              </Button>
            </DialogActions>
          </Dialog>

          <Paper
            className={
              this.state.selectedPlugin === ''
                ? this.props.classes.fullPaper
                : this.props.classes.leftPaper
            }
            elevation={4}
          >
            <Typography variant="headline" component="h3">
              Plugins
            </Typography>

            <List component="nav" className={this.props.classes.pluginsList}>
              {this.props.plugins.map((plugin, index) => (
                <ListItem
                  className={
                    plugin === this.state.selectedPlugin ? this.props.classes.selectedPlugin : null
                  }
                  key={plugin}
                  button
                  onClick={() => {
                    this.handlePluginSelect(plugin);
                  }}
                >
                  <ListItemIcon>{pollerIcon(plugin)}</ListItemIcon>
                  <ListItemText primary={plugin} />
                </ListItem>
              ))}
            </List>
          </Paper>
          {this.state.selectedPlugin !== '' && (
            <Paper className={this.props.classes.rightPaper}>
              <Typography variant="headline" component="h3" className={this.props.classes.title}>
                Details
              </Typography>
              <form className={this.props.classes.container} noValidate autoComplete="off">
                <TextField
                  id="name"
                  label="Name"
                  onChange={this.handleChange('name')}
                  className={this.props.classes.textField}
                  value={this.state.name}
                  margin="normal"
                />
                <TextField
                  id="description"
                  label="Description"
                  onChange={this.handleChange('description')}
                  className={this.props.classes.textField}
                  value={this.state.description}
                  margin="normal"
                />
                <TextField
                  id="coefficients"
                  label="Coefficients"
                  onChange={this.handleChange('coefficients')}
                  className={this.props.classes.textField}
                  value={this.state.coefficients}
                  margin="normal"
                />
                <TextField
                  id="station"
                  label="Station"
                  onChange={this.handleChange('station')}
                  className={this.props.classes.textField}
                  value={this.state.station}
                  margin="normal"
                />
                <TextField
                  id="pollRate"
                  label="Poll Rate"
                  onChange={this.handleChange('pollRate')}
                  className={this.props.classes.textField}
                  value={this.state.pollRate}
                  margin="normal"
                />
                <TextField
                  id="pin"
                  label="Pin"
                  onChange={this.handleChange('pin')}
                  className={this.props.classes.textField}
                  value={this.state.pin}
                  margin="normal"
                />
                <TextField
                  id="units"
                  label="Units"
                  onChange={this.handleChange('units')}
                  className={this.props.classes.textField}
                  value={this.state.units}
                  margin="normal"
                />
                <TextField
                  id="endpoint"
                  label="Endpoint"
                  onChange={this.handleChange('endpoint')}
                  className={this.props.classes.textField}
                  value={this.state.endpoint}
                  margin="normal"
                />
                <TextField
                  id="meta"
                  label="Meta"
                  onChange={this.handleChange('meta')}
                  className={this.props.classes.textField}
                  value={this.state.meta}
                  margin="normal"
                />
                <Button className={this.props.classes.button} onClick={testSensor}>
                  Test Sensor
                </Button>
                <Button disabled className={this.props.classes.button}>
                  Submit Sensor
                </Button>
              </form>
            </Paper>
          )}
        </div>

        <img
          className={this.props.classes.image}
          src="/screenshots/home.webp"
          id="dragonfly-image"
          alt="Home screen of Dragonfly"
          onClick={fullscreen}
          onKeyDown={fullscreen}
        />
        <p className={this.props.classes.descriptionContainer}>
          Dragonfly is all purpose sensor poller designed to be an easy interface for using
          raspberry pis. You can see in the picture on the right an example of what dragonfly would
          look like with multiple sensors hooked up. Upon hooking up a sensor to a raspberry pi, the
          user can then link a poller to it and add polling details like poll rate, coefficients, a
          station name and description. If the user has many pis running at the same time, multiple
          dragonfly instances can be linked to each other or forward all of their data to a central
          server. Once a sensor has been hooked up, actions can then be added to to the poller like
          sending an email or slack or turning on another sensor
          <br />
          The example above allows you to see some of the plugins available. The only one
          functioning is the cryptopoller. In the metadata field, you can add the name of your
          favorite cryptocurrency and it will respond with its current price in US dollars.
        </p>
      </div>
    );
  }
}

export default withStyles(styles)(DragonflyPage);
