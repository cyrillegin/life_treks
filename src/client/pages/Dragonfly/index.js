import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MonetizationOnIcon from '@material-ui/icons/MonetizationOn';
import ACUnitIcon from '@material-ui/icons/AcUnit';
import ShowChartIcon from '@material-ui/icons/ShowChart';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import DialogTitle from '@material-ui/core/DialogTitle';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContentText from '@material-ui/core/DialogContentText';
import './styles.scss';

const PLUGINS = ['cryptoPoller', 'customEntry', 'gpioPoller', 'BMP180Poller', 'DHT11Poller'];

export default class Dragonfly extends Component {
  // static propTypes = {
  //   testPlugin: PropTypes.func.isRequired,
  // };

  constructor(props) {
    super(props);
    this.state = {
      dialogIsOpen: false,
      dialogMessage: '',
      dialogTitle: '',
      selectedPlugin: '',
      pollRate: '',
      name: '',
      description: '',
      coefficients: '',
      station: '',
      pin: '',
      units: '',
      endpoint: '',
      meta: '',
    };
  }

  handlePluginSelect(name) {
    this.setState({
      selectedPlugin: name,
    });
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  };

  testSensor = () => {
    this.testPlugin(this.state.selectedPlugin, {
      name: this.state.name,
      description: this.state.description,
      coefficients: this.state.coefficients,
      station: this.state.station,
      pollRate: this.state.pollRate,
      pin: this.state.pin,
      units: this.state.units,
      endpoint: this.state.endpoint,
      meta: this.state.meta,
    }).then(data => {
      this.setState({
        dialogIsOpen: true,
        dialogMessage: data,
        dialogTitle: 'Test Result',
      });
    });
  };

  closeDialog = () => {
    this.setState({
      dialogIsOpen: false,
    });
  };

  pollerIcon = icon => {
    switch (icon) {
      case 'cryptoPoller':
        return <MonetizationOnIcon />;
      case 'gpioPoller':
        return <ACUnitIcon />;
      default:
        return <ShowChartIcon />;
    }
  };

  fullscreen = e => {
    const image = document.getElementById('dragonfly-image');
    if (image.requestFullscreen) {
      image.requestFullscreen();
    } else if (image.webkitRequestFullscreen) {
      image.webkitRequestFullscreen();
    } else if (image.mozRequestFullScreen) {
      image.mozRequestFullScreen();
    } else if (image.msRequestFullscreen) {
      image.msRequestFullscreen();
    } else {
      console.error('Fullscreen API is not supported.');
    }
  };

  testPlugin = (plugin, details) =>
    new Promise((res, rej) => {
      if (plugin === 'cryptoPoller') {
        fetch('/crypto', {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({ plugin, details }),
        })
          .then(response => response.json())
          .then(data => {
            res(data);
          });
      } else {
        res({});
      }
    });

  render() {
    return (
      <div className="dragonfly-root">
        <div className="demo-container">
          <Dialog
            open={this.state.dialogIsOpen}
            onClose={this.closeDialog}
            onClick={event => {
              event.nativeEvent.stopImmediatePropagation();
            }}
          >
            <DialogTitle>{this.state.dialogTitle}</DialogTitle>
            <DialogContent>
              <DialogContentText>{this.state.dialogMessage}</DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button onClick={this.closeDialog} color="primary">
                Okay
              </Button>
            </DialogActions>
          </Dialog>

          <Paper
            className={this.state.selectedPlugin === '' ? 'full-paper' : 'left-paper'}
            elevation={4}
          >
            <Typography variant="headline" component="h3">
              Plugins
            </Typography>

            <List component="nav" className="plugins-list">
              {PLUGINS.map((plugin, index) => (
                <ListItem
                  className={plugin === this.state.selectedPlugin ? 'selected-plugin' : null}
                  key={plugin}
                  button
                  onClick={() => {
                    this.handlePluginSelect(plugin);
                  }}
                >
                  <ListItemIcon>{this.pollerIcon(plugin)}</ListItemIcon>
                  <ListItemText primary={plugin} />
                </ListItem>
              ))}
            </List>
          </Paper>
          {this.state.selectedPlugin !== '' && (
            <Paper className="right-paper">
              <Typography variant="headline" component="h3" className="title">
                Details
              </Typography>
              <form className="container" noValidate autoComplete="off">
                <TextField
                  id="name"
                  label="Name"
                  onChange={this.handleChange('name')}
                  className="text-field"
                  value={this.state.name}
                  margin="normal"
                />
                <TextField
                  id="description"
                  label="Description"
                  onChange={this.handleChange('description')}
                  className="text-field"
                  value={this.state.description}
                  margin="normal"
                />
                <TextField
                  id="coefficients"
                  label="Coefficients"
                  onChange={this.handleChange('coefficients')}
                  className="text-field"
                  value={this.state.coefficients}
                  margin="normal"
                />
                <TextField
                  id="station"
                  label="Station"
                  onChange={this.handleChange('station')}
                  className="text-field"
                  value={this.state.station}
                  margin="normal"
                />
                <TextField
                  id="pollRate"
                  label="Poll Rate"
                  onChange={this.handleChange('pollRate')}
                  className="text-field"
                  value={this.state.pollRate}
                  margin="normal"
                />
                <TextField
                  id="pin"
                  label="Pin"
                  onChange={this.handleChange('pin')}
                  className="text-field"
                  value={this.state.pin}
                  margin="normal"
                />
                <TextField
                  id="units"
                  label="Units"
                  onChange={this.handleChange('units')}
                  className="text-field"
                  value={this.state.units}
                  margin="normal"
                />
                <TextField
                  id="endpoint"
                  label="Endpoint"
                  onChange={this.handleChange('endpoint')}
                  className="text-field"
                  value={this.state.endpoint}
                  margin="normal"
                />
                <TextField
                  id="meta"
                  label="Meta"
                  onChange={this.handleChange('meta')}
                  className="text-field"
                  value={this.state.meta}
                  margin="normal"
                />
                <Button className="button" onClick={this.testSensor}>
                  Test Sensor
                </Button>
                <Button disabled className="button">
                  Submit Sensor
                </Button>
              </form>
            </Paper>
          )}
        </div>
        <div role="button" onClick={this.fullscreen} onKeyPress={this.fullscreen} tabIndex={0}>
          <img
            className="image"
            src="/screenshots/home.webp"
            id="dragonfly-image"
            alt="Home screen of Dragonfly"
          />
        </div>
        <p className="description-container">
          Dragonfly is all purpose sensor poller designed to be an easy interface for using
          raspberry pis. You can see in the picture on the right an example of what dragonfly would
          look like with multiple sensors hooked up. Upon hooking up a sensor to a raspberry pi, the
          user can then link a poller to it and add polling details like poll rate, coefficients, a
          station name and description. If the user has many pis running at the same time, multiple
          dragonfly instances can be linked to each other or forward all of their data to a central
          server. Once a sensor has been hooked up, actions can then be added to to the poller like
          sending an email or slack or turning on another sensor
          <br />
          The example above allows you to see some of the plugins available. The only one
          functioning is the cryptopoller. In the metadata field, you can add the name of your
          favorite cryptocurrency and it will respond with its current price in US dollars.
        </p>
      </div>
    );
  }
}
