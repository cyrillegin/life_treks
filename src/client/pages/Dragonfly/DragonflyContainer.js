import {compose, mapProps} from 'recompose';
import Dragonfly from './Dragonfly';

export default compose(
  mapProps((ownProps) => {
    return {
      ...ownProps,
    };
  }),
)(Dragonfly);
