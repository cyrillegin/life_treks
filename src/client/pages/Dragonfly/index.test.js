import React from 'react';
import renderer from 'react-test-renderer';
import Dragonfly from './index';

describe('Dragonfly', () => {
  it('should render snapshots', () => {
    const tree = renderer
      .create(<Dragonfly classes={{}} testPlugin={() => {}} plugins={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
