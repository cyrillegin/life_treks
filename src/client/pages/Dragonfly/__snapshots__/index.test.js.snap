// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragonfly should render snapshots 1`] = `
<div
  className="dragonfly-root"
>
  <div
    className="demo-container"
  >
    <div
      className="MuiPaper-root-17 MuiPaper-elevation4-23 MuiPaper-rounded-18 full-paper"
    >
      <h3
        className="MuiTypography-root-44 MuiTypography-headline-49"
      >
        Plugins
      </h3>
      <nav
        className="MuiList-root-80 MuiList-padding-81 plugins-list"
      >
        <div
          className="MuiButtonBase-root-96 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-92 MuiListItem-button-93"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-99"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-100"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1.41 16.09V20h-2.67v-1.93c-1.71-.36-3.16-1.46-3.27-3.4h1.96c.1 1.05.82 1.87 2.65 1.87 1.96 0 2.4-.98 2.4-1.59 0-.83-.44-1.61-2.67-2.14-2.48-.6-4.18-1.62-4.18-3.67 0-1.72 1.39-2.84 3.11-3.21V4h2.67v1.95c1.86.45 2.79 1.86 2.85 3.39H14.3c-.05-1.11-.64-1.87-2.22-1.87-1.5 0-2.4.68-2.4 1.64 0 .84.65 1.39 2.67 1.91s4.18 1.39 4.18 3.91c-.01 1.83-1.38 2.83-3.12 3.16z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-109"
          >
            <span
              className="MuiTypography-root-44 MuiTypography-subheading-51 MuiListItemText-primary-112"
            >
              cryptoPoller
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-115"
          />
        </div>
        <div
          className="MuiButtonBase-root-96 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-92 MuiListItem-button-93"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-99"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-100"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-109"
          >
            <span
              className="MuiTypography-root-44 MuiTypography-subheading-51 MuiListItemText-primary-112"
            >
              customEntry
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-115"
          />
        </div>
        <div
          className="MuiButtonBase-root-96 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-92 MuiListItem-button-93"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-99"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-100"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-109"
          >
            <span
              className="MuiTypography-root-44 MuiTypography-subheading-51 MuiListItemText-primary-112"
            >
              gpioPoller
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-115"
          />
        </div>
        <div
          className="MuiButtonBase-root-96 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-92 MuiListItem-button-93"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-99"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-100"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-109"
          >
            <span
              className="MuiTypography-root-44 MuiTypography-subheading-51 MuiListItemText-primary-112"
            >
              BMP180Poller
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-115"
          />
        </div>
        <div
          className="MuiButtonBase-root-96 MuiListItem-root-84 MuiListItem-default-87 MuiListItem-gutters-92 MuiListItem-button-93"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemIcon-root-99"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-100"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root-109"
          >
            <span
              className="MuiTypography-root-44 MuiTypography-subheading-51 MuiListItemText-primary-112"
            >
              DHT11Poller
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-115"
          />
        </div>
      </nav>
    </div>
  </div>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
  >
    <img
      alt="Home screen of Dragonfly"
      className="image"
      id="dragonfly-image"
      src="/screenshots/home.webp"
    />
  </div>
  <p
    className="description-container"
  >
    Dragonfly is all purpose sensor poller designed to be an easy interface for using raspberry pis. You can see in the picture on the right an example of what dragonfly would look like with multiple sensors hooked up. Upon hooking up a sensor to a raspberry pi, the user can then link a poller to it and add polling details like poll rate, coefficients, a station name and description. If the user has many pis running at the same time, multiple dragonfly instances can be linked to each other or forward all of their data to a central server. Once a sensor has been hooked up, actions can then be added to to the poller like sending an email or slack or turning on another sensor
    <br />
    The example above allows you to see some of the plugins available. The only one functioning is the cryptopoller. In the metadata field, you can add the name of your favorite cryptocurrency and it will respond with its current price in US dollars.
  </p>
</div>
`;
